day_two_guide/topics/scaling_etcd.adoc

```diff
@@ -0,0 +1,432 @@
+for more information.
+====
+
+. Create the directory where the configuration and certificates are stored:
++
+----
+# mkdir -p ${PREFIX}
+----
+
+. Create the server certificate request and sign it:
++
+----
+# openssl req -new -config ${OPENSSLCFG} \
+    -keyout ${PREFIX}server.key  \
+    -out ${PREFIX}server.csr \
+    -reqexts etcd_v3_req -batch -nodes \
+    -subj /CN=$CN
+
+# openssl ca -name etcd_ca -config ${OPENSSLCFG} \
+    -out ${PREFIX}server.crt \
+    -in ${PREFIX}server.csr \
+    -extensions etcd_v3_ca_server -batch
+----
+
+. Create the peer certificate request and sign it:
++
+----
+# openssl req -new -config ${OPENSSLCFG} \
+    -keyout ${PREFIX}peer.key \
+    -out ${PREFIX}peer.csr \
+    -reqexts etcd_v3_req -batch -nodes \
+    -subj /CN=$CN
+
+# openssl ca -name etcd_ca -config ${OPENSSLCFG} \
+  -out ${PREFIX}peer.crt \
+  -in ${PREFIX}peer.csr \
+  -extensions etcd_v3_ca_peer -batch
+----
+
+. Copy the current etcd configuration and `ca.crt` files from the current node
+as examples to be modified later:
++
+----
+# cp /etc/etcd/etcd.conf ${PREFIX}
+# cp /etc/etcd/ca.crt ${PREFIX}
+----
+
+. Add the new host to the etcd cluster. Note the new host is not configured yet
+so the status stays as `unstarted` until the new host is properly configured:
++
+----
+# etcdctl2 member add ${NEW_ETCD_HOSTNAME} https://${NEW_ETCD_IP}:2380
+----
++
+This command outputs the following variables:
++
+----
+ETCD_NAME="<NEW_ETCD_HOSTNAME>"
+ETCD_INITIAL_CLUSTER="<NEW_ETCD_HOSTNAME>=https://<NEW_HOST_IP>:2380,<CLUSTERMEMBER1_NAME>=https:/<CLUSTERMEMBER2_IP>:2380,<CLUSTERMEMBER2_NAME>=https:/<CLUSTERMEMBER2_IP>:2380,<CLUSTERMEMBER3_NAME>=https:/<CLUSTERMEMBER3_IP>:2380"
+ETCD_INITIAL_CLUSTER_STATE="existing"
+----
+
+. Those values must be overwritten by the current ones in the sample
+`${PREFIX}/etcd.conf` file. Also, modify the following variables with the new
+host IP (`${NEW_ETCD_IP}` can be used) in that file:
++
+----
+ETCD_LISTEN_PEER_URLS
+ETCD_LISTEN_CLIENT_URLS
+ETCD_INITIAL_ADVERTISE_PEER_URLS
+ETCD_ADVERTISE_CLIENT_URLS
+----
+
+. Modify the `${PREFIX}/etcd.conf` file and check for syntax errors or missing
+IPs otherwise the etcd service could fail:
++
+----
+# vi ${PREFIX}/etcd.conf
+----
+
+. Once the file has been properly modified, a `tgz` file with the certificates,
+the sample configuration file, and the `ca` is created and copied to the new
+host:
++
+----
+# tar -czvf /etc/etcd/generated_certs/${CN}.tgz -C ${PREFIX} .
+# scp /etc/etcd/generated_certs/${CN}.tgz ${CN}:/tmp/
+----
+
+*Steps to be performed on the new etcd host*
+
+The new host is required to be subscribed to the proper Red Hat software
+channels as explained above in the prerequisites section.
+
+[discrete]
+== Procedure
+
+. Install `iptables-services` to provide iptables utilities to open the required
+ports for etcd:
++
+----
+# yum install -y iptables-services
+----
+
+. Create firewall rules to allow etcd to communicate:
```

Comment:
maybe

. Create the `OS_FIREWALL_ALLOW` filewall rules to allow etcd to communicate with:

* Clients on port 2379/tcp
* etcd peers on port 2380/tcp

the note about the chain